### PR TITLE
Add Prometheus Custom Metrics

### PR DESCRIPTION
**Pull request by dustman9000**
This adds prometheus custom metrics in order to have some observability of the operator via on-cluster prometheus.
Example of new metrics added:
```
# HELP must_gather_operator_must_gather_errors Number of must gathers with errors
# TYPE must_gather_operator_must_gather_errors counter
must_gather_operator_must_gather_errors 1
# HELP must_gather_operator_must_gather_total Total number of must gathers performed
# TYPE must_gather_operator_must_gather_total counter
must_gather_operator_must_gather_total 2
```

### CODE REVIEW

File: deploy/04_prometheus-k8s_role.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (2uasimojo) [inline deploy/04_prometheus-k8s_role.yaml:1]: Hm, why do we need a Role that gives the same permissions as the ClusterRole above, but in a specific namespace? That's not redundant?

File: pkg/controller/mustgather/mustgather_controller.go [hunk 2]
```diff
@@ -216,9 +217,11 @@ func (r *ReconcileMustGather) Reconcile(request reconcile.Request) (reconcile.Re
 
 	if err != nil {
 		if errors.IsNotFound(err) {
+			// Increment prometheus metrics for must gather total
+			localmetrics.MetricMustGatherTotal.Inc()
```
Comment (2uasimojo) [inline pkg/controller/mustgather/mustgather_controller.go:221]: Do we want to wait to do this until after the error check?

**Review by 2uasimojo — COMMENTED** [review]
This looks lovely. A couple of questions inline.

File: deploy/04_prometheus-k8s_role.yaml (hunk 1 above)
Comment (fahlmant) [reply deploy/04_prometheus-k8s_role.yaml:1]: This role is for the openshift-monitoring serviceaccount `prometheus-k8s`. It needs these perms, but it shouldn't be given globally for just one operator.

File: pkg/controller/mustgather/mustgather_controller.go (hunk 2 above)
Comment (dustman9000) [reply pkg/controller/mustgather/mustgather_controller.go]: Good point. Moved to after job creation.

File: deploy/04_prometheus-k8s_role.yaml (hunk 1 above)
Comment (dustman9000) [reply deploy/04_prometheus-k8s_role.yaml:1]: Yeah, this needs to scoped to the namespace.

Comment (2uasimojo) [reply deploy/04_prometheus-k8s_role.yaml:1]: Okay, I get it now. This bit allows read access for the monitoring service for scraping; the clusterrole above is bound to the service account for the operator itself.

**Review by 2uasimojo — APPROVED** [review]
/lgtm

**Pull request merged**